test/runtests.jl

```diff
@@ -30,4 +50,7 @@ if HDF5.has_parallel()
     include("mpio.jl")
 end
 
+# Clean up after all resources
+HDF5.API.h5_close()

```

Comment:
Could you create a new PR with this change or edit this commit to remove this line and add it as  a new commit. (It helps to have the commits logically separated and in this case the debug statements from the resource clean up fix).

Reply:
Usually for something like this  a git rebase interactive is helpful.  `git rebase -i HEAD~6` then go  to this commit and edit it (you can also rearrange the commits etc.) 

The reason I'm asking is I'd like to merge this PR without squashing the PR since there are several logically independent commits in the PR, which are useful to preserve in the git history.